.github/workflows/release.yml

```diff
@@ -9,7 +9,87 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+  tests-conan:
+    uses: ./.github/workflows/test_conan.yml
+  check-dockerfile:
+    uses: ./.github/workflows/check_dockerfile.yml
+  build-conan:
+    needs: tests-conan
+    strategy:
+      fail-fast: true
+      matrix:
+        config:
+          - { os: ubuntu-22.04, cc: "gcc", cxx: "g++" }
+          - { os: macos-14, cc: "clang", cxx: "clang++" }
+
+    runs-on: ${{ matrix.config.os }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Restore ccache caches
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: ${{ github.job }}
+
+      - name: Conan cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.conan/data
+          key: conan-${{ runner.os }}-${{ hashFiles('**/conanfile.py') }}
+
+      - name: apt cache
+        uses: actions/cache@v4
+        if: runner.os == 'Linux'        
+        with:
+          path: |
+            /var/cache/apt/
+            /var/lib/apt/
+          key: apt-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/install_deps_ubuntu.sh') }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+
+      - name: Set up pipx
+        run: |
+          python -m pip install --user pipx
+          python -m pipx ensurepath
+
+      - name: Set up Conan
+        run: |
+          pipx install conan
+
+      - name: Install compilers (Ubuntu)
+        if: runner.os == 'Linux'
+        run: sudo apt install build-essential ninja-build ccache cmake
```

Comment:
but ubuntu 22.04 does not have cmake 3.24+

Reply:
Good catch. I've used 24.04 in this CI initially so it made sense. I think I'm going to install cmake using pipx in this workflow

Reply:
@NikitaZotov changed macOS version only

Reply:
Fixed